aptos-move/framework/aptos-framework/sources/account/common_account_abstractions/ethereum_derivable_account.move

```diff
@@ -0,0 +1,429 @@
+/// Derivable account abstraction that verifies a message signed by
+/// SIWE.
+/// 1. The message format is as follows:
+///
+/// <domain> wants you to sign in with your Ethereum account:
+/// <ethereum_address>
+///
+/// To execute transaction <entry_function_name> on Aptos blockchain
+/// (<network_name>).
+///
+/// URI: <domain>
+/// Version: 1
+/// Chain ID: <chain_id>
+/// Nonce: <digest>
+/// Issued At: <issued_at>
+///
+/// 2. The abstract public key is a BCS serialized `SIWEAbstractPublicKey`.
+/// 3. The abstract signature is a BCS serialized `SIWEAbstractSignature`.
+/// 4. This module has been tested for the following wallets:
+/// - Metamask
+module aptos_framework::ethereum_derivable_account {
+    use aptos_framework::auth_data::AbstractionAuthData;
+    use aptos_std::secp256k1;
+    use aptos_std::option;
+    use aptos_std::aptos_hash;
+    use std::bcs_stream::{Self, deserialize_u8};
+    use std::chain_id;
+    use std::string_utils;
+    use std::transaction_context::{Self, EntryFunctionPayload};
+    use std::vector;
+    use std::string::String;
+
+    /// Signature failed to verify.
+    const EINVALID_SIGNATURE: u64 = 1;
+    /// Entry function payload is missing.
+    const EMISSING_ENTRY_FUNCTION_PAYLOAD: u64 = 2;
+    /// Invalid signature type.
+    const EINVALID_SIGNATURE_TYPE: u64 = 3;
+    /// Address mismatch.
+    const EADDR_MISMATCH: u64 = 4;
+    /// Unexpected v value.
+    const EUNEXPECTED_V: u64 = 5;
+
+    enum SIWEAbstractSignature has drop {
+        EIP1193DerivedSignature {
+            signed_chain_id: u8,
+            issued_at: String,
```

Comment:
what are these two? should they be part of public_key instead? i.e. if signed_chain_id is different, we should get a different account?

also they are not used/checked anywhere?

Reply:
`signed_chain_id` will be removed soon, it was part of the initial implementation on the adapter side but we dont need it anymore.

`issued_at` is a required field in the SIWE message, so we include it in the signature the client sends so we can build the message with it